### ISSUE TITLE
Weight of banners in Zone Probability screen is confusing

### ISSUE THREAD

**Issue by erikgeurts** [revive-adserver/revive-adserver#411]
The weight being displayed in the Probability tab of a zone can be confusing. Let's say the zone has 2 banners linked to it, each for a separate campaign. Campaign weight is set to 5 in each case. The banner weight of the banner from campaign A is set to 5, the weight of the banner from campaign B is the default value of 1.

In the zone probability, this results in the weight of the banner of campaign A being displayed as 25 (5 multiplied by 5), and the weight of the banner of campaign B is displayed as 5.

![zone-probability](https://cloud.githubusercontent.com/assets/5102461/4268647/d85e6f58-3cb1-11e4-8cf3-955285e23c7c.jpg)

Probability of each banner is still 50% since the weight of the 2 campaigns the banners come from is equal.

I believe many years ago, probability was calculated based on the relative weights of the multiplication results, but this has been changed to just take campaign weight into account.

**Comment by hsuttong** [comment]
Anyone know the reason negative banner weights are allowed? When looking at the probability of a neg. weight banner it just makes it a negative percentage. 😳

**Comment by mbeccati** [comment]
@hsuttong that looks like an input validation bug. Would you mind opening a new ticket for it?

**Comment by hsuttong** [comment]
@mbeccati created issue: #711

**Comment by Neurogami** [comment]
What seems to be the case (and what I hope is more than just coincidentally true for my experimenting) is that probabilities are effectively calculated twice: The odds that one or another campaign is shown, and then, for whatever campaign is selected, the odds of one or another banner being shown. 

For example, suppose I have a single ad slot on my site, and have two advertisers, each with one campaign. 

Advertiser 1 has one ad; advertiser 2 has two banners. 

Advertiser 2 has weighted their banners as 1 and 2; the second banner should be preferred 2 out 3 times.

Each campaign has a weight of 1.  That means there's a  50/50 chance of seeing an ad from either of the advertisers.

If campaign/advertiser 1 gets selected then the only ad associated gets shown; that banner's weight is irrelevant.

If campaign/advertiser 2 gets selected then the associated banner weights are examined.  Banner 2 should be selected to fill the campaign slot twice as often as banner 1.

If one were to calculate these odds in advance, we would expect advertiser 1's single ad to appear on the Web site 50% of the time.  Advertiser 2's first ad would appear about 16.6% of the time, and the other ad 33.% of the time.  

In other words, banner weight only relates to the odds of a given ad being selected among other ads in the same campaign,  and only after that campaign has been select (based on campaign weight).    Since advertiser 1 has a single banner ad, the weight of that ad has no bearing on its odds of being shown; it's the only ad in that campaign that *can* be shown, when that campaign has been selected.

**Comment by erikgeurts** [comment]
It seems you wrote what I wrote 2 years ago, but with a lot more words :)

**Comment by Neurogami** [comment]
Nothing on this  issue helped clarify for me ether the specific intended  or actual behavior, so I tried to write an explicit description of what I was seeing, and my understanding of it.    (Much of my experience with Revive so far has been having to experiment to see what the actual behavior is.)

**Comment by erikgeurts** [comment]
For questions and answers, we prefer to use our community forums at http://forum.revive-adserver.com

**Comment by Neurogami** [comment]
I commented here because a thread on that community forum about weights linked to this bug.  That thread didn't answer my question, the bug is still open (after almost 2 years)  and other users are  likely to be as confused as I was about what behavior to expect.  And, like me, users may find the existing comments on this open bug only marginally helpful  

In lieu of the bug being fixed perhaps such comments can guide people in the right direction.

**Comment by erikgeurts** [comment]
It's not a bug in my opinion.

**Comment by carolynjkruger** [comment]
@Neurogami Your response above was very helpful in trying to understand how the probability function works.
